AdminServer/appscale/admin/__init__.py

```diff
@@ -429,22 +381,35 @@ def delete(self, project_id):
       project_id: The id of the project to delete.
     """
     self.authenticate(project_id, self.ua_client)
-    project_path = constants.PROJECT_NODE_TEMPLATE.format(project_id)
-    update_project_state(self.zk_client, project_id,
-                         LifecycleState.DELETE_REQUESTED)
-    ports_to_close = []
-    # Delete each version of each service of the project.
-    for service_id in \
-        self.zk_client.get_children("{0}/services".format(project_path)):
-      for version_id in self.zk_client.get_children(
-          "{0}/services/{1}/versions".format(project_path, service_id)):
+    raise CustomHTTPError(HTTPCodes.NOT_IMPLEMENTED,
+                          message='Project deletion is not supported')
+
+
+class ServicesHandler(BaseVersionHandler):
+  """ Manages a project's services. """
+  def initialize(self, ua_client, zk_client):
+    self._ua_client = ua_client
+    self._zk_client = zk_client
+
+  def get(self, project_id):
+    """ Lists all the services in a project. """
+    self.authenticate(project_id, self._ua_client)
+    project_node = '/'.join(['/appscale', 'projects', project_id])
+    services_node = '/'.join([project_node, 'services'])
+    if not self._zk_client.exists(project_node):
+      raise CustomHTTPError(HTTPCodes.NOT_FOUND,
+                            message='Project does not exist')
 
-        port = yield self.start_delete_version(project_id, service_id,
-                                               version_id)
-        ports_to_close.append(port)
+    try:
+      service_ids = self._zk_client.get_children(services_node)
+    except NoNodeError:
+      raise CustomHTTPError(HTTPCodes.INTERNAL_ERROR,
+                            message='Services node not found for project')
 
-    IOLoop.current().spawn_callback(self.wait_for_delete,
-                                    ports_to_close, project_id)
+    prefix = '/'.join(['apps', project_id, 'services'])
```

Comment:
Probably not the best PR to comment about this, but should we be under /appscale or something like that instead of /apps?

Reply:
This is the response body of the Admin API call. The service representation is defined here: https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps.services#Service